proto/rpc.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+import "account.proto";
+import "block.proto";
+import "digest.proto";
+import "note.proto";
+import "transaction.proto";
+
+message GetBlockHeaderRequest {
+    BlockSpec block_spec = 1;
```

Comment:
The original discussion has this comment:

```
On the response side, how were you thinking about partially materialized blocks? Or more generally, should we be able to specify which parts of the block to return? There could be a few options here:

    By default we always get just the block header.
    We could also request to get a "compacted block" which would contain such things as hashes of created notes, consumed nullifiers, hashes of updated accounts, proof etc. - but would not contain more detailed data about public notes/accounts.
    We could request to get a full block: same as above + all detailed data about public notes/accounts.
    We could specify various parts of the block we are interested in. For example, get hashes of all created notes and updated accounts, but data about proof or nullifiers (not sure if such a request would be useful - just listing it as an example).
    We could specify some more sophisticated filters. For example, get block header + notes which match a certain tag (together with their authentication paths).
```

The recommended approach for protobuf/gRPC APIs is to use a field set for this. The idea is that the request includes a list of the data it needs, and the server only replies with that. References:

- [Include a Field Read Mask in Read Requests](https://protobuf.dev/programming-guides/api/#include-field-read-mask)
- [field_mask.proto](https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/field_mask.proto)

We can support the above with a single endpoint `GetBlockDataRequest` and a mask.